plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryMetadata.java

```diff
@@ -236,6 +246,41 @@ private List<SchemaTableName> listTablesInRemoteSchema(BigQueryClient client, St
         return tableNames.build();
     }
 
+    @Override
+    public Iterator<RelationCommentMetadata> streamRelationComments(ConnectorSession session, Optional<String> schemaName, UnaryOperator<Set<SchemaTableName>> relationFilter)
+    {
+        if (isLegacyMetadataListing) {
+            return ConnectorMetadata.super.streamRelationComments(session, schemaName, relationFilter);
+        }
+
+        BigQueryClient client = bigQueryClientFactory.create(session);
+        List<String> schemaNames = schemaName.map(name -> {
+            DatasetId localDatasetId = client.toDatasetId(name);
+            String remoteSchemaName = getRemoteSchemaName(client, localDatasetId.getProject(), localDatasetId.getDataset());
+            return List.of(remoteSchemaName);
+        }).orElseGet(() -> listSchemaNames(session));
+        Map<SchemaTableName, RelationCommentMetadata> resultsByName = schemaNames.stream()
+                .flatMap(schema -> {
```

Comment:
this method is super complex and hard to read. Can we split it into smaller methods? maybe extract some utility methods that handle those 404 errors?